test/functional/CompletionSpec.hs

```diff
@@ -250,6 +250,67 @@ spec = describe "completions" $ do
         item ^. insertTextFormat `shouldBe` Just Snippet
         item ^. insertText `shouldBe` Just "mapM ${1:a -> m b} ${2:t a}"
 
+    it "work for infix functions" $ runSession hieCommand fullCaps "test/testdata/completion" $ do
+      doc <- openDoc "Completion.hs" "haskell"
+      _ <- skipManyTill loggingNotification (count 2 noDiagnostics)
+
+      let te = TextEdit (Range (Position 5 7) (Position 5 24)) "even `filte"
+      _ <- applyEdit doc te
+
+      compls <- getCompletions doc (Position 5 18)
+      let item = head $ filter ((== "filter") . (^. label)) compls
+      liftIO $ do
+        item ^. label `shouldBe` "filter"
+        item ^. kind `shouldBe` Just CiFunction
+        item ^. insertTextFormat `shouldBe` Just Snippet
+        item ^. insertText `shouldBe` Just "filter`"
```

Comment:
Can we add some assertions about the placeholders the function is suggesting after completing? It would be painful if the placeholders we suggest contain the left hand side of the function

Reply:
The placeholders are the expected type? e.g. in the case of filter: `filter' {2:[a]}`?

Reply:
With backticks the placeholders should just drop the first element right? Because we can assume that the left hand side will supply the first argument

Reply:
I dont know. Is this really wanted? I find myself often writing such functions in exisiting expressions, normally I dont want these placeholders...

Reply:
This is something that has crossed my mind too, but I think it needs a more general solution, to include in the context if there is anything to the right already, and in that case not offering placeholders

Reply:
Maybe I am also misunderstanding the question. This particular completion omits the placeholders completely at the moment.

Reply:
So you are proposing that we keep it like that?  Which could work. If anyone feels strongly about it we can add them again. I do think that infix operations tend to be binary, so the missing placeholder(s) is probable not an issue.

Reply:
Yes, I think it would be better and we can add it anytime.